leaf-ui/rappid-extensions/BackboneModels.js

```diff
@@ -1 +1,455 @@
+     * and a repFuncSegment containing three FuncSegments inside of it,
+     * this function returns an array of size 4, containing all the
+     * function segments in chronological order.
+     *
+     * Every FuncSegment in the returned array will be a deep copy of the
+     * origin FuncSegments.
+     *
+     * Each FuncSegment in the returned array will have a inRepeat attribute
+     * which evaluates to true iff the FuncSegment was part of a
+     * repFunctionSegment
+     *
+     * @returns {Array.<FuncSegment>}
+     */
+    // TODO update this function - very unsure abt code/logic here
+    // Add all FuncSegmentBBMs to array and all FuncSegmentBBMs in RepFuncSegmentBBM
+    // To the same array in chronilogical order
+    getFuncSegmentIterable: function() {
+        var res [];
+        var repFuncSegmentList = this.RepFuncSegment.get('functionSegList');
+        var repStartTime = repFuncSegmentList[0].get('startTP');
+        // iterates over all of the FuncSegmentBBMs and adds them to new array
+        for (var i = 0; i < this.functionSegList.length; i++) {
+            var obj = this.functionSegList[i].clone();
+            // if the start the current funcSegmentBBM's start time is before the first 
+            // start time of the RepFuncSegmentBBMs, add it to array
+            if (this.functionSegList[i].get('startTP') <= repStartTime) {
+            res.push(obj);
+            }
+            // if its larger, add all of the FuncSegementBBMs in the RepFuncSegmentBBM
+            // to the array first
+            else {
+                if (RepFuncSegment != null) {
+                    for (var i = 0; i < this.RepFuncSegment.get('functionSegList').length; i++) {
+                        var obj = this.functionSegList[i].clone();
+                        res.push(obj);
+                    }   
+                }
+                res.push(obj); 
+            }
+        }   
+        return res;       
+        // Original function
+        // var res = [];
+        // for (var i = 0; i < this.functionSegList.length; i++) {
+        //     var obj = this.functionSegList[i];
+        //     if (obj instanceof FuncSegmentModel) {
+        //         var clone = Object.assign(new FuncSegment, obj); // deep copy
+        //         clone.isRepeat = false;
+        //         res.push(obj);
+        //     } else {
+        //         var segList = obj.functionSegList;
+        //         for (var j = 0; j < segList.length; j++) {
+        //             var clone = Object.assign(new FuncSegment, segList[j]);
+        //             clone.isRepeat = true;
+        //             res.push(clone);
+        //         }
+        //     }
+        // }
+        // return res;
+    }, 
+
+    /**
+     * Returns true if this EvolvingFunction contains
+     * a repeating segment (ie, contains a RepFuncSegment)
+     *
+     * @returns {Boolean}
+     */
+    hasRepeat: function() {
+        if (RepFuncSegment != null) {

```

Comment:
```suggestion
        return (RepFuncSegment != null)
```

Reply:
I think I renamed this to be `RepFunctionSegment`? Please confirm...

Reply:
Yes, I'll update this now